client/lib/query-manager/activity/index.js

```diff
@@ -29,8 +29,14 @@ export default class ActivityQueryManager extends QueryManager {
 	 * @return {Number}       0 if equal, less than 0 if itemA is first,
 	 *                        greater than 0 if itemB is first.
 	 */
-	static compare( query, { activityTs: tsA }, { activityTs: tsB } ) {
-		return Date.parse( tsB ) - Date.parse( tsA );
```

Comment:
Dropping this `Date.parse` looked suspicious:

https://github.com/Automattic/wp-calypso/blob/37520163ca8373fbcb35fb0a0e3d68f720eb4a48/client/state/data-layer/wpcom/sites/activity/from-api.js#L69

Turns out there was likely a bug:

```js
Date.parse( '2018-04-05T15:06:03.884Z' )
// 1522940763884
Date.parse( Date.parse( '2018-04-05T15:06:03.884Z' ) )
// NaN 😱 Whoops!
[3,2,1].sort(()=>NaN)
// [ 3, 2, 1 ]
```

I'd guess we weren't even sorting 😬

Reply:
>  I don't think this will address any of the recent sequencing issues

Maybe related? I bet a lot of that was hidden by the fact that previously 1 month corresponded to 1 query, so API-ordered activity was fine.